modules/auxiliary/server/capture/smb.rb

```diff
@@ -25,626 +29,250 @@ def initialize
-          win_name,
-          dns_domain,
-          dns_name,
-          @challenge,
-          sb_flag
-        )
-      else
-        securityblob = NTLM_UTILS::make_ntlmssp_blob_chall(
-          win_domain,
-          win_name,
-          dns_domain,
-          dns_name,
-          @challenge,
-          sb_flag
-        )
+      File.open(path, 'ab') do |f|
+        f.puts(creds[:combined_hash])
       end
-      pkt['Payload'].v['SecurityBlobLen'] = securityblob.length
-      pkt['Payload'].v['Payload'] = securityblob
-
-      c.put(pkt.to_s)
-
-    when NTLM_MESSAGE::Type3
-      # we can process the hash and send a status_logon_failure response packet
-
-      # Record the remote multiplex ID
-      smb[:multiplex_id] = pkt['Payload']['SMB'].v['MultiplexID']
-      lm_len = ntlm_message.lm_response.length # Always 24
-      nt_len = ntlm_message.ntlm_response.length
-
-      if nt_len == 24 # lmv1/ntlmv1 or ntlm2_session
-        arg = {
-          :ntlm_ver => NTLM_CONST::NTLM_V1_RESPONSE,
-          :lm_hash => ntlm_message.lm_response.unpack('H*')[0],
-          :nt_hash => ntlm_message.ntlm_response.unpack('H*')[0]
-        }
-
-        if @s_ntlm_esn && arg[:lm_hash][16,32] == '0' * 32
-          arg[:ntlm_ver] = NTLM_CONST::NTLM_2_SESSION_RESPONSE
-        end
-        # if the length of the ntlm response is not 24 then it will be
-        # bigger and represent an NTLMv2 response
-      elsif nt_len > 24 # lmv2/ntlmv2
-        arg = {
-          :ntlm_ver => NTLM_CONST::NTLM_V2_RESPONSE,
-          :lm_hash => ntlm_message.lm_response[0, 16].unpack('H*')[0],
-          :lm_cli_challenge => ntlm_message.lm_response[16, 8].unpack('H*')[0],
-          :nt_hash => ntlm_message.ntlm_response[0, 16].unpack('H*')[0],
-          :nt_cli_challenge => ntlm_message.ntlm_response[16, nt_len - 16].unpack('H*')[0]
-        }
-      elsif nt_len == 0
-        print_status("SMB Capture - Empty hash from #{smb[:name]} - #{smb[:ip]} captured, ignoring ... ")
-        smb_error(CONST::SMB_COM_SESSION_SETUP_ANDX, c, CONST::SMB_STATUS_LOGON_FAILURE, true)
-        return
-      else
-        print_status("SMB Capture - Unknown hash type from #{smb[:name]} - #{smb[:ip]}, ignoring ...")
-        smb_error(CONST::SMB_COM_SESSION_SETUP_ANDX, c, CONST::SMB_STATUS_LOGON_FAILURE, true)
-        return
-      end
-
-      buff = pkt['Payload'].v['Payload']
-      buff.slice!(0,securityblobLen)
-      names = buff.split("\x00\x00").map { |x| x.gsub(/\x00/, '') }
-
-      smb[:username] = ntlm_message.user
-      smb[:domain]   = ntlm_message.domain
-      smb[:peer_os]  = names[0]
-      smb[:peer_lm]  = names[1]
+    end
 
-      begin
-        smb_get_hash(smb,arg,true)
-      rescue ::Exception => e
-        print_error("SMB Capture - Error processing Hash from #{smb[:name]} - #{smb[:ip]} : #{e.class} #{e} #{e.backtrace}")
+    # Cain & Abel doesn't support import of NTLMv2 hashes
+    if datastore['CAINPWFILE'] && creds[:jtr_format] == 'netntlm'
```

Comment:
Looks like this constant reference was missed.

```suggestion
    if datastore['CAINPWFILE'] && creds[:jtr_format] == JTR_NTLMV1
```